x-pack/plugins/security_solution/server/lib/timeline/routes/notes/get_notes.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import type { SecuritySolutionPluginRouter } from '../../../../types';
+import { NOTE_URL } from '../../../../../common/constants';
+
+import type { ConfigType } from '../../../..';
+import type { SetupPlugins } from '../../../../plugin';
+
+import { buildSiemResponse } from '../../../detection_engine/routes/utils';
+import { buildFrameworkRequest, getNotesPaginated } from '../../utils/common';
+import { getAllSavedNote } from '../../saved_object/notes';
+import { noteSavedObjectType } from '../../saved_object_mappings/notes';
+
+export const getNotesByDocumentIdsRoute = (
+  router: SecuritySolutionPluginRouter,
+  _: ConfigType,
+  security: SetupPlugins['security']
+) => {
+  router.versioned
+    .get({
+      path: NOTE_URL,
+      options: {
+        tags: ['access:securitySolution'],
+      },
+      access: 'public',
+    })
+    .addVersion(
+      {
+        validate: {
+          request: { query: getNotesPaginated },
+        },
+        version: '2023-10-31',
+      },
+      async (context, request, response) => {
+        try {
+          const queryParams = request.query;
+          const frameworkRequest = await buildFrameworkRequest(context, security, request);
+          const alertIds = queryParams.alertIds ?? null;
+          if (alertIds != null) {
+            if (Array.isArray(alertIds)) {
+              const alertIdSearchString = alertIds?.join(' | ');
+              const options = {
+                type: noteSavedObjectType,
+                search: alertIdSearchString,
+              };
+              const res = await getAllSavedNote(frameworkRequest, options);
+
+              return response.ok({ body: res ?? {} });
+            } else {
+              const options = {
+                type: noteSavedObjectType,
+                search: alertIds,
+              };
+              const res = await getAllSavedNote(frameworkRequest, options);
+
+              return response.ok({ body: res ?? {} });
+            }
+          } else {
+            const perPage = queryParams?.perPage ? parseInt(queryParams.perPage, 10) : 10;
```

Comment:
If this can return just all notes, I would rename this controller to `getNotesRoute` instead of `getNotesByDocumentIdsRoute`